src/tests/class_tests/openms/source/ConsensusMapMergerAlgorithm_test.cpp

```diff
@@ -0,0 +1,103 @@
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Julianus Pfeuffer $
+// $Authors: Julianus Pfeuffer $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+#include <OpenMS/ANALYSIS/ID/ConsensusMapMergerAlgorithm.h>
+#include <OpenMS/test_config.h>
+#include <OpenMS/FORMAT/ConsensusXMLFile.h>
+
+using namespace OpenMS;
+using namespace std;
+
+START_TEST(ConsensusMapMergerAlgorithm, "$Id$")
+
+    START_SECTION(mergeAllIDRuns)
+      {
+        ConsensusXMLFile cf;
+        ConsensusMap cmap;
+        cf.load(OPENMS_GET_TEST_DATA_PATH("BSA.consensusXML"), cmap);
+        ConsensusMapMergerAlgorithm cmerge;
+        cmerge.mergeAllIDRuns(cmap);
+        TEST_EQUAL(cmap.getProteinIdentifications().size(), 1)
+      }
+    END_SECTION
+
+    START_SECTION(mergeProteinsAcrossFractionsAndReplicates (no Design))
+      {
+        ConsensusXMLFile cf;
+        ConsensusMap cmap;
+        cf.load(OPENMS_GET_TEST_DATA_PATH("BSA.consensusXML"), cmap);
+        ConsensusMapMergerAlgorithm cmerge;
+        ExperimentalDesign ed = ExperimentalDesign::fromConsensusMap(cmap);
+        cmerge.mergeProteinsAcrossFractionsAndReplicates(cmap, ed);
+        //without a special experimental design on sample level, runs are treated like replicates
+        // or fractions and all are merged
+        TEST_EQUAL(cmap.getProteinIdentifications().size(), 1)
+        StringList toFill; cmap.getProteinIdentifications()[0].getPrimaryMSRunPath(toFill);
+        TEST_EQUAL(toFill.size(), 6)
+      }
+    END_SECTION
+
+    START_SECTION(mergeProteinsAcrossFractionsAndReplicates)
+      {
+        ConsensusXMLFile cf;
+        ConsensusMap cmap;
+        cf.load(OPENMS_GET_TEST_DATA_PATH("BSA.consensusXML"), cmap);
+        ConsensusMapMergerAlgorithm cmerge;
+        ExperimentalDesign ed = ExperimentalDesign::fromConsensusMap(cmap);
+        ExperimentalDesign::SampleSection ss{
+            {{"1","C1"},{"2","C2"},{"3","C3"}},
+            {{1,0},{2,1},{3,2}},
+            {{"Sample",0},{"Condition",1}}
+        };
+        ed.setSampleSection(ss);
+        cmerge.mergeProteinsAcrossFractionsAndReplicates(cmap, ed);
+        TEST_EQUAL(cmap.getProteinIdentifications().size(), 3)
+        StringList toFill; cmap.getProteinIdentifications()[0].getPrimaryMSRunPath(toFill);
+        TEST_EQUAL(toFill.size(), 2)
+        TEST_EQUAL(toFill[0], "/Users/pfeuffer/git/OpenMS-inference-src/share/OpenMS/examples/FRACTIONS/BSA1_F1.mzML")
+        TEST_EQUAL(toFill[1], "/Users/pfeuffer/git/OpenMS-inference-src/share/OpenMS/examples/FRACTIONS/BSA1_F2.mzML")
```

Comment:
@jpfeuffer this is not going to work...

Reply:
Hmm.. why not? you mean the absolute paths? In this case the absolute paths will work since the primaryMSRunPath represents the original mzML files with which the runs in the consensusMap were generated and they will never change through the pipeline.
Or do you mean something else?

Reply:
that is what I meant, I thought that the file will be missing on our test machines ... but I guess its okay?

Reply:
Right. No, the files won't be present on the build machines. But as it is implemented now, the existence is only checked when really needed (e.g. to look up spectra during writing mzTab) otherwise they are carried along as-is (e.g. also during merging like here).